tests/DateScalarTest.php

```diff
@@ -0,0 +1,130 @@
+<?php declare(strict_types=1);
+
+namespace MLL\GraphQLScalars\Tests;
+
+use DateTimeImmutable;
+use Generator;
+use GraphQL\Error\Error;
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Language\AST\IntValueNode;
+use GraphQL\Language\AST\StringValueNode;
+use MLL\GraphQLScalars\DateScalar;
+use PHPUnit\Framework\TestCase;
+
+abstract class DateScalarTest extends TestCase
+{
+    /**
+     * @dataProvider invalidDateValues
+     *
+     * @param mixed $value An invalid value for a date
+     */
+    public function testThrowsIfSerializingInvalidDates($value): void
+    {
+        $this->expectException(InvariantViolation::class);
+
+        $this->scalarInstance()->serialize($value);
+    }
+
+    /**
+     * @dataProvider invalidDateValues
+     *
+     * @param mixed $value An invalid value for a date
+     */
+    public function testThrowsIfParseValueInvalidDate($value): void
+    {
+        $this->expectException(Error::class);
+
+        $this->scalarInstance()->parseValue($value);
+    }
+
+    /**
+     * Those values should fail passing as a date.
+     *
+     * @return Generator<array-key, array{mixed}>
+     */
+    public function invalidDateValues(): Generator
+    {
+        yield [1];
+        yield ['rofl'];
+        yield [null];
+        yield [''];
+        yield ['2020-02-30'];
```

Comment:
I don't think this should be a valid value for `DateTime`. Let's split this provider into cases that are general and cases that are specific.

Reply:
it's in invalid provider (?)

Should I add a comment title above?

Reply:
Oops, yeah that makes sense - it can definitely stay there.

What do you think about adding an additional test to ensure that, for example:
- `DateTime` does not accept `2020-02-03`
- `Date` does not accept `2020-02-03 12:34:56`

Reply:
what about now

Reply:
awesome!